### PR TITLE
Move get_unwrap to restriction

### LINKED ISSUE #3862: get_unwrap is questionable advice
The [get_unwrap](https://rust-lang.github.io/rust-clippy/master/index.html#get_unwrap) lint suggests replacing `x.get(y).unwrap()` with `x[y]`. IMO, this is bad advice because the former syntax makes it clear that there's a risk of a panic, while the latter looks innocuous. It's generally good to be aware of where panics can happen. It's especially true when processing untrusted input, where a panic caused by user input is a Denial-of-Service vulnerability.

I propose that we make this lint non-default at the least, and possibly consider removing it.

### PR DESCRIPTION
**Pull request by Manishearth**
fixes #3862

r? @oli-obk

### CODE REVIEW

**Comment by Manishearth** [comment]
Haven't actually gotten a chance to test this

File: tests/ui/get_unwrap.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 // run-rustfix
 #![allow(unused_mut)]
+#![deny(get_unwrap)]
```
Comment (phansch) [inline tests/ui/get_unwrap.rs]:
```suggestion
#![deny(clippy::get_unwrap)]
```
That should fix the test failure

**Comment by Manishearth** [comment]
Fixed

**Pull request merged**